src/sentry/integrations/notifications.py

```diff
@@ -48,15 +42,21 @@ def get_channel_and_integration_by_user(
         # recipients.
         return {}
 
+    identity_id_to_idp: Mapping[APIIdentity.id, APIIdentityProvider | None] = {
```

Comment:
That's really smart and I didn't know that mypy supported this.  I like `Kind` ordered logic and this is pretty nifty!